apps/api/src/simulation-run/simulation-run.service.ts

```diff
@@ -792,68 +795,88 @@ export class SimulationRunService {
+
+            for (const format of BIOSIMULATIONS_FORMATS) {
+              if (
+                format?.biosimulationsMetadata?.modelFormatMetadata?.sedUrn &&
+                model.language.startsWith(
+                  format?.biosimulationsMetadata?.modelFormatMetadata?.sedUrn,
+                )
+              ) {
+                modelFormat = format;
+                break;
+              }
+            }
+            
+            const algorithmKisaoId =
+              taskAlgorithmMap[uri] || simulation.algorithm.kisaoId;
+            const algorithmKisaoTerm = this.ontologiesService.getOntologyTerm(
+              Ontologies.KISAO,
+              algorithmKisaoId,
+            );
 
-          summaryTasks.push({
-            uri: docLocation + '/' + task.id,
-            id: task.id,
-            name: task?.name,
-            model: {
-              uri: docLocation + '/' + task.model.id,
-              id: task.model.id,
-              name: task.model?.name,
-              source: task.model.source,
-              language: {
-                name: modelFormat?.name || undefined,
-                acronym:
-                  modelFormat?.biosimulationsMetadata?.acronym || undefined,
-                sedmlUrn: task.model.language,
-                edamId: modelFormat?.id || undefined,
-                url: modelFormat?.url || undefined,
-              },
-            },
-            simulation: {
-              type: {
-                id: task.simulation._type,
-                name: SimulationTypeName[task.simulation._type],
-                url: 'https://sed-ml.org/',
+            summaryTasks.push({
+              uri: docLocation + '/' + task.id,
+              id: task.id,
+              name: task?.name,
+              model: {
+                uri: docLocation + '/' + model.id,
+                id: model.id,
+                name: model?.name,
+                source: model.source,
+                language: {
+                  name: modelFormat?.name || undefined,
+                  acronym:
+                    modelFormat?.biosimulationsMetadata?.acronym || undefined,
+                  sedmlUrn: model.language,
+                  edamId: modelFormat?.id || undefined,
+                  url: modelFormat?.url || undefined,
+                },
               },
-              uri: uri,
-              id: task.simulation.id,
-              name: task.simulation?.name,
-              algorithm: {
-                kisaoId: algorithmKisaoId,
-                name:
-                  algorithmKisaoTerm?.name ||
-                  `${algorithmKisaoId} (deprecated)`,
-                url:
-                  algorithmKisaoTerm?.url ||
-                  'https://www.ebi.ac.uk/ols/ontologies/kisao',
+              simulation: {
+                type: {
+                  id: simulation._type,
+                  name: SimulationTypeName[simulation._type as keyof typeof SimulationTypeName],
```

Comment:
Is there a way to add the type for `_type` to the parent structure so we don't need to use as?